app/models/concerns/with_profile.rb

```diff
@@ -4,7 +4,7 @@ module WithProfile
   module ClassMethods
     def for_profile(profile)
       where(uid: profile.uid).first_or_initialize.tap do |user|
-        user.assign_attributes(profile.to_h)
+        user.assign_attributes(profile.to_h.delete_if {|_,v| v.nil?})
         user.save!
```

Comment:
Why not just using `compact`?

Reply:
crap, I thought hashes didn't understand compact